cupy/_core/_gufuncs.py

```diff
@@ -191,15 +192,15 @@ def _process_signatures(self, signatures):
             ops.append(_OpsRegister._Op(ins, outs, op))
         return ops
 
-    def _determine_from_args(self, args, casting):
+    def _determine_from_args(self, args):
```

Comment:
Forgot to comment: Casting is not relevant if we are looking for the right loop (promotion), safe-casting must be used for things to make sense.

As far as I can tell, this code is unused though.  GUfuncs just wrap the "default_func" always currently.